### PR TITLE
PP-5300 Refactor DD connector to remove unused fields on GET Payment

### PR DESCRIPTION
**Pull request by RoryMalcolm**
- The response from getCharge() in PaymentsResource now maps
  onto what is expected in publicAPI
- Changes the return type from PaymentResponse to
  CollectPaymentResponse

### CODE REVIEW

File: src/main/java/uk/gov/pay/directdebit/payments/services/PaymentService.java [hunk 1]
```diff
@@ -113,36 +105,21 @@ Payment submitPaymentToProvider(Payment payment) {
...
-                    GET,
-                    nextUrl(directDebitConfig.getLinks().getFrontendUrl(), "secure", token.getToken())));
-            dataLinks.add(createLink("next_url_post",
-                    POST,
-                    nextUrl(directDebitConfig.getLinks().getFrontendUrl(), "secure"),
-                    APPLICATION_FORM_URLENCODED,
-                    ImmutableMap.of("chargeTokenId", token.getToken())));
-        }
-        return PaymentResponse.from(payment, dataLinks);
+    public CollectPaymentResponse createPaymentResponseWithAllLinks(Payment payment) {
```
Comment (oswaldquek) [inline src/main/java/uk/gov/pay/directdebit/payments/services/PaymentService.java]: There's no point to this method anymore as links aren't set on the payment. This method can be deleted.

File: src/test/java/uk/gov/pay/directdebit/payments/resources/PaymentResourceIT.java [hunk 2]
```diff
@@ -285,59 +285,12 @@ public void shouldRetrieveAPayment_fromPublicApiEndpoint() {
...
                 .statusCode(OK.getStatusCode())
                 .contentType(JSON)
-                .body(JSON_CHARGE_KEY, is(paymentFixture.getExternalId()))
                 .body(JSON_AMOUNT_KEY, isNumber(paymentFixture.getAmount()))
                 .body(JSON_REFERENCE_KEY, is(paymentFixture.getReference()))
                 .body(JSON_DESCRIPTION_KEY, is(paymentFixture.getDescription()))
                 .body(JSON_STATE_STATUS_KEY, is(paymentFixture.getState().toExternal().getStatus()))
                 .body(JSON_STATE_FINISHED_KEY, is(false))
-                .body(JSON_STATE_DETAILS_KEY, is("example_details"))
-                .body(JSON_RETURN_URL_KEY, is(mandateFixture.getReturnUrl()));
```
Comment (oswaldquek) [inline src/test/java/uk/gov/pay/directdebit/payments/resources/PaymentResourceIT.java]: the fields JSON_RETURN_URL_KEY, JSON_CHARGE_KEY and FRONTEND_CARD_DETAILS_URL can be deleted. imports can be optimized on this class too.

File: src/test/java/uk/gov/pay/directdebit/payments/services/PaymentServiceTest.java [hunk 3]
```diff
@@ -140,15 +138,13 @@ public void shouldCreateACollectPaymentResponseFromAValidTransaction() {
     }
     @Test
     public void shouldCreateATransactionResponseWithLinksFromAValidTransaction() {
-        when(mockedTokenService.generateNewTokenFor(mandateFixture.toEntity())).thenReturn(new Token("token", mandateFixture.getId()));
-        PaymentResponse paymentResponse = service
-                .createPaymentResponseWithAllLinks(paymentFixture.toEntity(),
-                        gatewayAccountFixture.getExternalId(), mockedUriInfo);
+        CollectPaymentResponse paymentResponse = service
+                .createPaymentResponseWithAllLinks(paymentFixture.toEntity()
+                );
```
Comment (oswaldquek) [inline src/test/java/uk/gov/pay/directdebit/payments/services/PaymentServiceTest.java]: the above three lines can be formatted better into one line: `var paymentResponse = service.createPaymentResponseWithAllLinks(paymentFixture.toEntity());`

**Comment by RoryMalcolm** [comment]
@oswaldquek Changes made 🙏

File: src/test/java/uk/gov/pay/directdebit/payments/services/PaymentServiceTest.java [hunk 4]
```diff
@@ -28,34 +27,22 @@
...
 
 import static org.hamcrest.CoreMatchers.notNullValue;
 import static org.hamcrest.core.Is.is;
 import static org.junit.Assert.assertThat;
-import static org.mockito.ArgumentMatchers.any;
-import static org.mockito.ArgumentMatchers.anyString;
-import static org.mockito.Mockito.verify;
-import static org.mockito.Mockito.verifyZeroInteractions;
-import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.*;
```
Comment (oswaldquek) [inline src/test/java/uk/gov/pay/directdebit/payments/services/PaymentServiceTest.java]: no wildcard imports https://gds-way.cloudapps.digital/manuals/programming-languages/java.html#imports

File: src/test/java/uk/gov/pay/directdebit/payments/services/PaymentServiceTest.java [hunk 5]
```diff
@@ -28,34 +27,22 @@
...
 import static uk.gov.pay.directdebit.payments.model.DirectDebitEvent.SupportedEvent.PAYMENT_SUBMITTED_TO_BANK;
 import static uk.gov.pay.directdebit.payments.model.Payment.PaymentBuilder.fromPayment;
-import static uk.gov.pay.directdebit.payments.model.PaymentState.CANCELLED;
-import static uk.gov.pay.directdebit.payments.model.PaymentState.EXPIRED;
-import static uk.gov.pay.directdebit.payments.model.PaymentState.FAILED;
-import static uk.gov.pay.directdebit.payments.model.PaymentState.NEW;
-import static uk.gov.pay.directdebit.payments.model.PaymentState.PENDING;
-import static uk.gov.pay.directdebit.payments.model.PaymentState.SUCCESS;
-import static uk.gov.pay.directdebit.payments.model.PaymentState.USER_CANCEL_NOT_ELIGIBLE;
+import static uk.gov.pay.directdebit.payments.model.PaymentState.*;
```
Comment (oswaldquek) [inline src/test/java/uk/gov/pay/directdebit/payments/services/PaymentServiceTest.java]: no wildcard imports

File: src/main/java/uk/gov/pay/directdebit/payments/services/PaymentService.java [hunk 6]
```diff
@@ -112,37 +104,14 @@ Payment submitPaymentToProvider(Payment payment) {
...
-    }
-
-    public CollectPaymentResponse collectPaymentResponseWithSelfLink(Payment payment, String accountExternalId, UriInfo uriInfo) {
-        List<Map<String, Object>> dataLinks = ImmutableList.of(
-                createLink("self", GET, selfUriFor(uriInfo, "/v1/api/accounts/{accountId}/charges/{paymentExternalId}",
-                        accountExternalId, payment.getExternalId()))
-        );
-        return CollectPaymentResponse.from(payment, dataLinks);
+    
+    public CollectPaymentResponse collectPaymentResponseFromPayment(Payment payment) {
```
Comment (oswaldquek) [inline src/main/java/uk/gov/pay/directdebit/payments/services/PaymentService.java]: this method can be deleted as its unused

File: src/test/java/uk/gov/pay/directdebit/payments/resources/PaymentResourceIT.java (hunk 2 above)
Comment (oswaldquek) [reply src/test/java/uk/gov/pay/directdebit/payments/resources/PaymentResourceIT.java]: this isn't resolved yet as https://github.com/alphagov/pay-direct-debit-connector/blob/master/src/test/java/uk/gov/pay/directdebit/payments/resources/PaymentResourceIT.java#L75, https://github.com/alphagov/pay-direct-debit-connector/blob/master/src/test/java/uk/gov/pay/directdebit/payments/resources/PaymentResourceIT.java#L67-L68 and https://github.com/alphagov/pay-direct-debit-connector/blob/master/src/test/java/uk/gov/pay/directdebit/payments/resources/PaymentResourceIT.java#L62 can be deleted. There are still some unused imports on this class.

**Pull request merged**